### ISSUE TITLE
Upgrade Protocol Handles to Follow the SPEC

### ISSUE THREAD

**Issue by whyrusleeping** [ipfs/kubo#1549]
To get multistream interop between node and go, we need to agree on the official protocol names.

stream muxers:
- spdystream -> `/stream-muxer/spdy/3.1.0`
- yamux -> `/stream-muxer/yamux/1.0.0`
- multiplex -> `/stream-muxer/multiplex/1.0.0`

ipfs protocols:
- ping -> `/ipfs/ping/1.0.0`
- dht -> `/ipfs/dht/1.0.0`
  - maybe use 'kademlia' in there somewhere
- bitswap -> `/ipfs/bitswap/1.0.0`
- diagnostics -> `/ipfs/diag/1.0.0`

**Comment by diasdavid** [comment]
It would be good to avoid using the name DHT, to avoid the Peer VS Content routing. Right now I'm calling the routing part that uses kademlia, kad-router, as a module name. so maybe we can have: /ipfs/router/kad/1.0.0, so that in the future we might have `/ipfs/router/anotherP2POverlayRouting/1.0.0` too.

For the record store, I haven't cored it yet, but probably  `/ipfs/record-store/1.0.0` would be good.

**Comment by whyrusleeping** [comment]
as far as I know the record store isnt a protocol.

**Comment by diasdavid** [comment]
@whyrusleeping here is a `ping` to remind you to support the protocol multicodecs by spec, as described on:
https://github.com/ipfs/specs/blob/master/libp2p/7-properties.md#757-protocol-multicodecs

**Comment by whyrusleeping** [comment]
depends on #3110

**Comment by Winterhuman** [comment]
@whyrusleeping @daviddias Should this be closed?

**Comment by gammazero** [comment]
This should be resolved now. See #3132

If there is any remaining problem, please open a new issue to correct/align a specific protocol name.
